sdk/sync/common.go

```diff
@@ -83,13 +84,13 @@ func MustWatcherWriter(runenv *runtime.RunEnv) (*Watcher, *Writer) {
 
 // WatcherWriter creates a Watcher and a Writer object associated with this test
 // run's sync tree.
-func WatcherWriter(runenv *runtime.RunEnv) (*Watcher, *Writer, error) {
-	watcher, err := NewWatcher(runenv)
+func WatcherWriter(ctx context.Context, runenv *runtime.RunEnv) (*Watcher, *Writer, error) {
```

Comment:
Same here; let's document that this context doesn't get stored and it's only used while connecting to the backend.